### DISCUSSION TITLE
QMediaPlayer volume issue on iOS

### DISCUSSION THREAD

**Discussion by i-mage** [quasarframework/quasar-ui-qmediaplayer#355] [General]
Anyone encountered an issue where volume cannot be set by using setVolume() method on iOS?

**Comment by i-mage** [comment]
Now I noticed that the volume slider doesn't work on iOS at all, only mute button is working.

**Comment by i-mage** [comment]
Could someone recommend a player that supports volume change on iOS in cordova mode?

**Comment by hawkeye64** [comment]
I have transferred this conversation to the proper repo.

The problem with Apple is they keep changing requirements. For volume, they no longer allow external software to change the volume. They expect the user to do it through the side buttons. However, I have heard there is a workaround. By outing the volume, then changing the volume, then unmuting. I have not tested this myself.

**Reply by i-mage** [reply]
I have noticed that even when playing YouTube in the browser on iOS you can't change the volume via the website. Will try the workaround, thanks!

**Reply by hawkeye64** [reply]
Silly Apple likes to lock things down so much. This is why they lose devs to their platform. :(

**Reply by i-mage** [reply]
> Silly Apple likes to lock things down so much. This is why they lose devs to their platform. :(

Yeah, it's incredible. They did this a couple of years ago already. Never even occurred to me that they would block something like volume change. I get it that the **main** volume is controlled by the user but within the app, you cannot even do some fade-outs, etc. no comment.

**Reply by hawkeye64** [reply]
I wonder how Spotify is doing it?

**Reply by i-mage** [reply]
> I wonder how Spotify is doing it?

I think Spotify is not based on Webaudio API or am I wrong?

**Reply by hawkeye64** [reply]
Actually, I have no idea.
